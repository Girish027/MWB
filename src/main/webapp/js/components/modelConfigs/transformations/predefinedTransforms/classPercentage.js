import Constants from 'constants/Constants';

const classPercentage = () => {
  const regex = {
    type: Constants.TRANSFORMATION_TYPES.WORDCLASS_SUBST_REGEX,
    mappings: {},

  };
  regex.mappings = {
    '/[0-9]+\\.[0-9]*\\s?(?:%|percent(?:age|ile)?)/i': '_class_percentage',
    '/[0-9]+\\s?(?:%|percent(?:age|ile)?)/i': '_class_percentage',
    '/(?:one(?:\\s|-))?hundred\\s?(?:%|percent(?:age|ile)?)/i': '_class_percentage',
    '/(?:\\b(?:twenty|thirty|forty|fifty|sixty|seventy|eighty|ninety)(?:(?:\\s|-)?(?:one|two|three|four|five|six|seven|eight|nine))?)\\s?(?:%|percent(?:age|ile)?)/i': '_class_percentage',
    '/\\b(?:zero|one|two|three|four|five|six|seven|eight|nine|hundred|ten|eleven|twelve|thirteen|fourteen|fifteen|sixteen|seventeen|eighteen|nineteen|twenty|thirty|forty|fifty|sixty|seventy|eighty|ninety)\\s?(?:%|percent(?:age|ile)?)/i': '_class_percentage',
  };

  return regex;
};

export default classPercentage;
