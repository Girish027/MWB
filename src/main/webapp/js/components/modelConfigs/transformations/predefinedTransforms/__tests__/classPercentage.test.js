import classPercentage from 'components/modelConfigs/transformations/predefinedTransforms/classPercentage';

describe('classPercentage', () => {
  const regex = classPercentage();

  test('type is equal to wordclass-subst-regex', () => {
    expect(regex.type).toEqual('wordclass-subst-regex');
  });

  test('mapping contains more than one key', () => {
    const keys = Object.keys(regex.mappings);
    expect(keys.length).toBeGreaterThan(1);
  });

  test('snapshot predefined transformation', () => {
    expect(regex.mappings).toMatchSnapshot();
  });
});
