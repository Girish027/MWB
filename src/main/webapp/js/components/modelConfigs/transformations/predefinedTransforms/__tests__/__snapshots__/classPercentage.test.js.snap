// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`classPercentage snapshot predefined transformation 1`] = `
Object {
  "/(?:\\\\b(?:twenty|thirty|forty|fifty|sixty|seventy|eighty|ninety)(?:(?:\\\\s|-)?(?:one|two|three|four|five|six|seven|eight|nine))?)\\\\s?(?:%|percent(?:age|ile)?)/i": "_class_percentage",
  "/(?:one(?:\\\\s|-))?hundred\\\\s?(?:%|percent(?:age|ile)?)/i": "_class_percentage",
  "/[0-9]+\\\\.[0-9]*\\\\s?(?:%|percent(?:age|ile)?)/i": "_class_percentage",
  "/[0-9]+\\\\s?(?:%|percent(?:age|ile)?)/i": "_class_percentage",
  "/\\\\b(?:zero|one|two|three|four|five|six|seven|eight|nine|hundred|ten|eleven|twelve|thirteen|fourteen|fifteen|sixteen|seventeen|eighteen|nineteen|twenty|thirty|forty|fifty|sixty|seventy|eighty|ninety)\\\\s?(?:%|percent(?:age|ile)?)/i": "_class_percentage",
}
`;
